pkg/pget.go

```diff
@@ -99,6 +104,10 @@ func (g *Getter) DownloadFiles(ctx context.Context, manifest Manifest) (int64, t
 	if err != nil {
 		return 0, 0, fmt.Errorf("error downloading files: %w", err)
 	}
+	err = g.Downloader.Wait()
+	if err != nil {
+		return 0, 0, fmt.Errorf("error downloading files: %w", err)
+	}
```

Comment:
I don't think this 100% solves the issue/panic. 

Double calls to .Wait() isn't the underlying issue. Since we execute the Fetches through g.Go() those happen in a wait group and if we call .Wait() here and deep in fetch we are slow and end up adding another element to the errgroup we'll see the same panic as before. 

Now this code does reduce the potential if the max concurrent files is < the length of the manifest as we would block on errgroup.Wait a few lines up. 

I think this points to a problem with complexity of using the errgroup as the semaphore and error channel.

Reply:
you're right that there's still a risk. I'll have another go. I think we might be missing an abstraction here.

Reply:
I've had a long think about this and I now think two things:

1. this PR actually is safe. We only call Wait() after every returned stream has been fed to consumer.Consume(), which means there really can't be any remaining goroutines reading to call errgroup.Go().
2. this is really really hard to think about, and I'm sure there's a simpler design inside screaming to get out.

So I think we should merge this PR as a way to fix #188, but I'm thinking hard about making this simpler overall.